beacon-chain/core/blocks/block_operations.go

```diff
@@ -84,11 +84,11 @@ func ProcessBlockRandao(
 	enableLogging bool,
 ) (*pb.BeaconState, error) {
 
-	proposerIdx, err := helpers.BeaconProposerIndex(beaconState, beaconState.Slot)
-	if err != nil {
-		return nil, fmt.Errorf("could not get beacon proposer index: %v", err)
-	}
 	if verifySignatures {
+		proposerIdx, err := helpers.BeaconProposerIndex(beaconState, beaconState.Slot)
+		if err != nil {
+			return nil, fmt.Errorf("could not get beacon proposer index: %v", err)
+		}
```

Comment:
Moved this into the verify signatures because it was kinda pointless being outside of it if the proposer index was only used for the verify randao.